mRides-app/PreferencesActivity.cs

```diff
@@ -98,30 +83,66 @@ protected override void OnCreate(Bundle savedInstanceState)
 
             // Set the done button to save and continue to the next activity
             Button doneButton = FindViewById<Button>(Resource.Id.buttonDone);
-            doneButton.Click += delegate { this.SaveAndContinue(rbSmoker.Checked, rbLuggage.Checked, rbHandicap.Checked, rbPet.Checked, (int)spinnerGender.SelectedItemId); };
+            string genderPreference = "";
+            int selectedGenderPref = (int)spinnerGender.SelectedItemId;
+            if (selectedGenderPref == 0)
+            {
+                genderPreference = "any";
+            }
+            else if (selectedGenderPref == 1)
+            {
+                genderPreference = "male";
+            }
+            else
+            {
+                genderPreference = "female";
+            }
+            doneButton.Click += delegate { this.SaveAndContinue(rbSmoker.Checked, rbLuggage.Checked, rbHandicap.Checked, rbPet.Checked, genderPreference); };
             
         }
 
-        private void SaveAndContinue(Boolean smoker, Boolean luggage, Boolean handicap, Boolean pet, int gender)
+        private void SaveAndContinue(Boolean smoker, Boolean luggage, Boolean handicap, Boolean pet, string gender)
         {
-            // TODO: Save the preferences
-            Console.WriteLine("Smoker=" + smoker + ";Luggage=" + luggage + ";handicap=" + handicap + ";gender=" + gender + ";pet=" + pet);
-            User.currentUser.isSmoker = smoker;
-            User.currentUser.hasLuggage = luggage;
-            User.currentUser.isHandicap = handicap;
-            User.currentUser.genderPreference = gender;
-            User.currentUser.hasPet = pet;
-            //User.currentUser.save();
-            
-            this.Continue();
-        }
+            // If the current user is null and the previous activity is the main, 
+            // create a new user using the preferences
+            string previousActivity = Intent.GetStringExtra(Constants.IntentExtraNames.PreviousActivity);
+            if (User.currentUser == null && previousActivity.Equals(Constants.ActivityNames.MainActivity)) {
+                long facebookID = Convert.ToInt64(Intent.GetStringExtra(Constants.IntentExtraNames.UserFacebookId));
+                string facebookFirstName = Intent.GetStringExtra(Constants.IntentExtraNames.UserFacebookFirstName);
+                string facebookLastName = Intent.GetStringExtra(Constants.IntentExtraNames.UserFacebookLastName);
+                User newUser = new User
```

Comment:
We can have a nondefault constructor of User that accepts all these fields, will make the code more compact and easier to read